### PR TITLE
Fix `group-columns` for `join-condition-rhs-columns`

### LINKED ISSUE #32509: [MLv2] [Bug] `group-columns` works incorrectly with `join-condition-rhs-columns`
When `join-condition-rhs-columns` are passed to `group-columns`, they get grouped into the source table instead of a join table.

### To Reproduce

Create a query from the sample Orders table and run this:

```js
const peopleTable = Lib.tableOrCardMetadata(5)
const columns = joinConditionRHSColumns(query, 0, peopleTable)
const [group] = ML.groupColumns(columns)

Lib.displayInfo(query, 0, group)

// Expected
{ name: "PEOPLE", displayName: "People", ...rest }

// Actual
{ name: "ORDERS", displayName: "Orders", ...rest }
```

### PR DESCRIPTION
**Pull request by camsaul**
Fixes #32509

### CODE REVIEW

File: src/metabase/lib/card.cljc [hunk 1]
```diff
@@ -20,14 +21,18 @@
   (cond-> card-metadata
     (not display-name) (assoc :display-name (u.humanization/name->human-readable-name :simple card-name))))
 
+(mu/defn fallback-display-name :- ::lib.schema.common/non-blank-string
```
Comment (camsaul) [inline src/metabase/lib/card.cljc:24]: Deduplicated this logic

File: src/metabase/lib/column_group.cljc [hunk 2]
```diff
@@ -39,59 +50,96 @@
   [_query _stage-number column-group]
   column-group)
 
+(defmulti ^:private display-info-for-group-method
```
Comment (camsaul) [inline src/metabase/lib/column_group.cljc:53]: The `display-info-method` below got to be pretty big with the new stuff I added for joins to it so I broke it out into a private multimethod with a few separate impls. It's a little easier on the eyes this way. I didn't change the code for the other stuff, only for joins.

File: src/metabase/lib/join.cljc [hunk 3]
```diff
@@ -620,10 +621,15 @@
    ;; I was on the fence about whether these should get `:lib/source :source/joins` or not -- it seems like based on
    ;; the QB UI they shouldn't. See screenshots in #31174
    (sort-join-condition-columns
-    (let [joinable (if (join? join-or-joinable)
+    (let [joinable   (if (join? join-or-joinable)
```
Comment (camsaul) [inline src/metabase/lib/join.cljc:624]: `join-condition-rhs-columns` should include the join alias if we're using it with an existing join; and it should include `:source/joins` so we trigger the join-group logic in `lib.column-group`

File: test/metabase/lib/test_util.cljc [hunk 4]
```diff
@@ -209,3 +209,42 @@
                                                     :base-type     :type/Integer
                                                     :semantic-type :type/FK}]}
                    :native             "SELECT whatever"}]})
+
+(def categories-mbql-card
```
Comment (camsaul) [inline test/metabase/lib/test_util.cljc:213]: Added several more helper mock Cards & metadata providers with them to facilitate the new tests and to help with other tests in the future.

File: src/metabase/lib/column_group.cljc [hunk 5]
```diff
@@ -39,59 +50,96 @@
...
+   ;; with [[metabase.lib.join/join-condition-rhs-columns]], the alias won't be present yet, so group things by the
+   ;; joinable -- either the Card we're joining, or the Table we're joining. Prefer `:lib/card-id` because when we
+   ;; join a Card the Fields might have `:table-id` but we want the entire Card to appear as one group. See #32493
+   (or
+    (when-let [join-alias (lib.join/current-join-alias column-metadata)]
+      {:join-alias join-alias})
+    (when card-id
+      {:card-id card-id})
+    (when table-id
+      {:table-id table-id}))))
```
Comment (camsaul) [inline src/metabase/lib/column_group.cljc:133]: The actual fix -- if we're grouping columns with `:source/joins` and it doesn't have a join alias (which is the case when using `join-condition-rhs-columns` when building a new join), we need to group by the Card ID or Table ID (the ID whatever it is we're joining against) instead. Then we can use this ID to add appropriate display-info for the group.

**Review by snoe — APPROVED** [review]

**Pull request merged**